src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ServiceUtilHelper.cs

```diff
@@ -199,6 +216,80 @@ public static void EnsureClientCertificateInstalled()
         ThrowIfClientCertificateInstallationError();
     }
 
+    // Tries to ensure that the server machine certificate is installed into
+    // the TrustedPeople store.  It obtains the server certificate from the service
+    // utility endpoint and either installs it or verifies that a matching
+    // one is already installed.  InvalidOperationException will be thrown
+    // if an error occurred attempting to install the certificate.  This
+    // method may be called multiple times but will attempt the installation
+    // once only.
+    public static void EnsureServerCertificateInstalled()
```

Comment:
@iamjasonp -- regarding passing a cert that can't chain build to our root CA cert.  I agree in principle, but looking at code coverage, I see the PeerTrust validator "wins" and we don't get to the code path that checks chain building.  In fact, to get coverage of that, we would need to pass in a cert that *fails* peer trust but *passes* chain trust. The only way I know how to do that is to pass in a cert not in the TrustedPeople store. Just FYI.